operator/duties/scheduler.go

```diff
@@ -279,13 +277,13 @@ func (s *Scheduler) SlotTicker(ctx context.Context) {
 
 			if waitDuration > 0 {
 				time.Sleep(waitDuration)
-
-				// Lock the mutex before broadcasting
-				s.waitCond.L.Lock()
-				s.headSlot = slot
-				s.waitCond.Broadcast()
-				s.waitCond.L.Unlock()
 			}
+
+			// Lock the mutex before broadcasting
+			s.waitCond.L.Lock()
+			s.headSlot = slot
+			s.waitCond.Broadcast()
+			s.waitCond.L.Unlock()
```

Comment:
Seems like we should advance `headSlot` regardless of whether or not **we actually had to wait** to 1/3 here (that is - if tick happened past 1/3 for whatever reason we still want to advance it, right) ?

Reply:
@iurii-ssv The only idea that comes to my mind as to why this could have been done like that is to make sure the slot is not decreased

Reply:
> @iurii-ssv The only idea that comes to my mind as to why this could have been done like that is to make sure the slot is not decreased

Not sure if that's even possible for "slot decreasing" to happen in the current code version (or how that time-based `if` would even help with that), but I'm adding [another change here](https://github.com/ssvlabs/ssv/pull/1860/files#r1850813217) that actually does make it a likely problem to happen,

and hence we probably want to have explicit protection against "slot decreasing" (which I've now added in this PR)